syntax = "proto3";

message User{
    string userName=1;
    string password=2;
    string email=3;
    int32 age=4;
    string location=5;
    string userId=6;
}

message UserList{
    repeated User users=1;
}


message Project{
    string name=1;
    string id=2; // uuid
    string userId=3;
    repeated string text=4;
}

message Notepad{
    repeated Project projects=1;
}

message Version{
    int32 changesCount=1;
    int32 versionNumber=2;
    string projectId=3;
    repeated string createdTime=4;
}

message Changes{
    enum change{
        ADD=0;
        REMOVE=1;
        UPDATE=2;
    }
    repeated int32 lineNum=1;
    repeated change action=2;
    repeated string changes=3;
    int32 changesCount=4;
    repeated string createdTime=5;
    repeated string id=6;
}